python/pyspark/sql/session.py

```diff
@@ -611,7 +611,7 @@ def _inferSchema(
         :class:`pyspark.sql.types.StructType`
         """
         first = rdd.first()
-        if not first:
+        if first != 0 and not first:
```

Comment:
I think we can just ignore this condition. If RDD is empty, `RDD.first` seems throwing an exception.

Reply:
Sorry I meant an RDD like `spark._sc.parallelize([[],[]])`, Spark SQL is able to create a DataFrame with an array column but PySpark raises an exception.

Reply:
Oh got it. What about we check if it is `collections.abc.Sized`, and if the size is `0` here instead?

Reply:
Even if we let it pass this if-else check, PySpark returns an empty DataFrame as below
```py
>>> spark.createDataFrame(spark._sc.parallelize([[],[]])).show()
++
||
++
||
||
++
```
whereas Spark SQL returns a DataFrame of empty arrays.
Let me check the code paths further and adjust that. Thanks!

Reply:
> Oh got it. What about we check if it is `collections.abc.Sized`, and if the size is `0` here instead?

+1 for this comment.

Otherwise looks good.